cmd/timestamp_check/types.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"time"
+
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/binance"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bitfinex"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bitflyer"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bithumb"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bitmex"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bitstamp"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/bittrex"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/btcmarkets"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/btse"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/coinbasepro"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/coinbene"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/coinut"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/exmo"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/gateio"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/gemini"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/hitbtc"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/huobi"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/itbit"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kraken"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/lakebtc"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/lbank"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/localbitcoins"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/okcoin"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/okex"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/poloniex"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/yobit"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/zb"
+)
+
+const defaultSleepTime = time.Minute
+
+// SupportedExchanges exchanges are supported exchanges
+var SupportedExchanges = []exchange.IBotExchange{
+	new(binance.Binance),
+	new(bitfinex.Bitfinex),
+	new(bitflyer.Bitflyer),
+	new(bithumb.Bithumb),
+	new(bitmex.Bitmex),
+	new(bitstamp.Bitstamp),
+	new(bittrex.Bittrex),
+	new(btcmarkets.BTCMarkets),
+	new(btse.BTSE),
+	new(coinbene.Coinbene),
+	new(coinut.COINUT),
+	new(exmo.EXMO),
+	new(coinbasepro.CoinbasePro),
+	new(gateio.Gateio),
+	new(gemini.Gemini),
+	new(hitbtc.HitBTC),
+	new(huobi.HUOBI),
+	new(itbit.ItBit),
+	new(kraken.Kraken),
+	new(lakebtc.LakeBTC),
+	new(lbank.Lbank),
+	new(localbitcoins.LocalBitcoins),
+	new(okcoin.OKCoin),
+	new(okex.OKEX),
+	new(poloniex.Poloniex),
+	new(yobit.Yobit),
+	new(zb.ZB),
+}
+
+const configFilename = "timestamp.json"
+
+// TimestampConfiguration defines the indidual exchange configuration
+type TimestampConfiguration struct {
+	Enabled bool `json:"enabled"`
+	Keys
+	Report *Report `json:"-"`
+}
+
+// Keys defines the individual access keys for the exchange
+type Keys struct {
```

Comment:
These should only get appended _if_ they're needed. You can check out how that works in exchange_wrapper_issues or elsewhere with `base.API.CredentialsValidator.RequiresClientID` etc